edit_book.php

```diff
@@ -0,0 +1,44 @@
+<?php
+include 'database.php';
+
+if (isset($_GET['id'])) {
```

Comment:
Consider adding unit tests for the update functionality. This will help ensure that the edit logic works as expected and can handle potential edge cases.

```php
use PHPUnit\Framework\TestCase;

class BookRepositoryTest extends TestCase {
    public function testUpdateBook() {
        $mockConnection = $this->createMock(mysqli::class);
        $mockStmt = $this->createMock(mysqli_stmt::class);

        $mockConnection->method('prepare')->willReturn($mockStmt);
        $mockStmt->expects($this->once())->method('bind_param')->with($this->equalTo('ssii'), $this->anything(), $this->anything(), $this->anything(), $this->anything());
        $mockStmt->method('execute')->willReturn(true);

        $repository = new BookRepository($mockConnection);
        $this->assertTrue($repository->updateBook(1, 'New Title', 'New Author', 2022));
    }
}
```